pkg/scalers/datadog_scaler.go

```diff
@@ -94,17 +87,24 @@ func parseDatadogMetadata(config *ScalerConfig) (*datadogMetadata, error) {
 	meta.query += rollup
 
 	if val, ok := config.TriggerMetadata["type"]; ok {
```

Comment:
I left the options to continue using the metadata.type entry to avoid a breaking change here, LMK if you think this should be aligned with the rest of the scalers.

Reply:
I think this scaler should align with the scalers. What if we start with making the datadog's `type` optional and deprecated with this change, with a warning message that users should start using the global `type`. If it is set.
And maybe left the possibilty to specify both options, but just check that both have the same value and error if not.
Datadog scaler is relatively new, so I think that we can deprecate this setting (and do the breaking change) in a few releases and don't have to wait for 3.x.

Reply:
And probably the same approach for CPU/Mem. WDYT @JorTurFer @tomkerkhove ?